lib/node_modules/@stdlib/math/generics/random/sample/lib/factory.js

```diff
@@ -20,35 +25,119 @@ var slice = Array.prototype.slice;
 /**
 * Returns a function to sample elements from an array-like object.
 *
-* @param {PositiveInteger} seed - integer-valued seed
+* @param {ArrayLike} [pool] - array-like object to sample from
+* @param {Options} [options] - function options
+* @param {PositiveInteger} [options.seed] - integer-valued seed
+* @param {NonNegativeInteger} [options.size] - sample size
+* @param {boolean} [options.replace=true] - boolean indicating whether to sample with replacement
 * @returns {Function} function to sample elements from an array-like object
 *
 * @example
-* var sample = factory( 323 );
+* var sample = factory({
+*     'seed': 232
+* });
+* var out = sample( 'abcdefg' );
+* // [ 'g', 'd', 'g', 'f', 'c', 'e', 'f' ]
+*
+* @example
+* var sample = factory( [1,2,3,4,5,6], {
+*     'seed': 232
+*     'size': 2
+* });
+* var out = sample();
+* // returns [ 6, 4 ]
+*
+* out = sample()
+* // returns [ 6, 5 ]
+*
+* @example
+* var sample = factory( [1,2,3,4,5,6], {
+*     'seed': 474
+*     'size': 3
+*     'replace': false
+* });
+* var out = sample();
+* // returns [ 4, 3, 6 ]
+*
+* out = sample()
+* // returns [ 1, 5, 2 ]
+*
+* out = sample()
+* // throws <Error>
+*
+* @example
+* var sample = factory( [0,1], {
+*     'size': 2
+* });
+*
+* var out = sample();
+* // e.g., returns [ 1, 1 ]
+*
+* out = sample({
+*     size: 10
+* });
+* // e.g, returns [ 0, 1, 1, 1, 0, 1, 0, 0, 1, 1 ]
+*
+* @example
+* var sample = factory( [0,1], {
+*     'size': 2
+* });
+*
+* var out = sample();
+* // e.g., returns [ 1, 1 ]
+*
+* out = sample({
+*     'replace': false
+*     'size': 1
+* });
+* // returns [ 0 ] or [ 1 ]
+*
+* out = sample();
+* // throws <RangeError>
```

Comment:
That is odd behavior. If I set the default `replace=true`, I have the expectation that I will be able to sample indefinitely. If I later set `replace=false` to temporarily override, causing my pool to shrink, thus affecting all subsequent calls, that, at least, violates my expectation, as I may reach a point where I can no longer draw. 

Instead, what I may expect is that my default behavior is sampling indefinitely, but every now and again I may want to temporarily override by sampling without replacement from the same pool. Essentially, this would allow me to reuse a pool without needing to provide a copy. I suppose internally, if the only way to sample without replacement is by mutating the pool, then we would have to copy to support this.

On the other hand, I could see a situation (not sure the use case) where someone wants to periodically shrink the larger pool size; i.e., every now and again take balls out of the urn. 

Either way, depending on user expectations, choosing one or approach or the other will be "surprising" to a user cohort.

Perhaps we should provide an option indicating which path a user wants?